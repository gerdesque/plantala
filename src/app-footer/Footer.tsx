import React from 'react';
import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';
import { Avatar, Grid } from '@material-ui/core';
import AvatarGroup from '@material-ui/lab/AvatarGroup';
import { IPlant } from '../plant/Plant';
import Drawer from '../drawer/Drawer';
import { avatarCount } from '../utils/Utils';

interface IFooterProps {
  selectedPlants: IPlant[],
  activePlant: IPlant,
  setActivePlant: any,
  transformPlant: any
}

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    footer: {
      padding: theme.spacing(2),
      backgroundColor: theme.palette.primary.main,
    },
    avatar: {
      width: theme.spacing(8),
      height: theme.spacing(8),
      borderColor: theme.palette.secondary.main,
      cursor: 'pointer'
    }
  }),
);

export default function Footer({selectedPlants, activePlant, setActivePlant, transformPlant}: IFooterProps) {
  const classes = useStyles();
  const avatars = selectedPlants.slice(0, avatarCount).map(plant =>
    <Avatar
      key={plant.name}
      alt={plant.name}
      src={`${process.env.PUBLIC_URL}/assets/${plant.source}`}
      className={classes.avatar}
      onClick={() => setActivePlant(plant)}
    />);

  return (
    <footer className={classes.footer}>
      <Grid container spacing={2} justify="center">
        <AvatarGroup max={avatarCount}>
          {avatars}
        </AvatarGroup>
        {activePlant && <Drawer
          activePlant={activePlant}
          setActivePlant={setActivePlant}
          transformPlant={transformPlant}
        />}
      </Grid>
    </footer>
  );
}
