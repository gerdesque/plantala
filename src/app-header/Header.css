.app-logo {
  height: 4em;
  width: 4em;
  margin-right: 1em;
}
