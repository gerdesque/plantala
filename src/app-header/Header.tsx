import React from 'react';
import { AppBar, IconButton, Toolbar, Typography } from '@material-ui/core';
import InvertColorsIcon from '@material-ui/icons/InvertColors';
import VolumeOffIcon from '@material-ui/icons/VolumeOff';
import VolumeUpIcon from '@material-ui/icons/VolumeUp';
import { makeStyles } from '@material-ui/core/styles';
import logo from '../assets/logo.webp';

interface IHeaderProps {
  setStart: any
  isPlaying: boolean
  setSound: any
  setColorMode: any
}

const useStyles = makeStyles((theme) => ({
  header: {
    backgroundColor: theme.palette.primary.main
  },
  icon: {
    marginRight: theme.spacing(2),
    height: theme.spacing(8),
    width: theme.spacing(8),
    cursor: 'pointer'
  },
  title: {
    flexGrow: 1,
  },
}));

export default function Header({ setStart, isPlaying, setSound, setColorMode }: IHeaderProps) {
  const classes = useStyles();

  return (
    <div>
      <AppBar position="relative" className={classes.header}>
        <Toolbar>
          <img src={logo} className={classes.icon} alt="Plantala logo" onClick={() => setStart()} />
          <Typography variant="h2" className={classes.title}>
            Plantala
          </Typography>
          <IconButton aria-label="invert" onClick={() => setSound(isPlaying)}>
            {isPlaying ? <VolumeUpIcon /> : <VolumeOffIcon />}
          </IconButton>
          <IconButton aria-label="invert" onClick={() => setColorMode()}>
            <InvertColorsIcon />
          </IconButton>
        </Toolbar>
      </AppBar>
    </div>
  )
}
