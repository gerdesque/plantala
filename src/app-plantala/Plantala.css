.plantala {
  height: 100vh;
  width: 100vw;
  display: grid;
  grid-template-rows: auto 1fr auto;
}

@media print {
  html, body { width: 210mm; height: 297mm;}
  footer, img, button, h5 { display: none !important; }
  .MuiAppBar-colorPrimary {box-shadow: none !important;}
  h2::before { content: "Malen. Strahlen. "; }
  h2::after { content: "!"; }
  .MuiContainer-root { max-width: 1600px !important; padding: 0 !important }
}