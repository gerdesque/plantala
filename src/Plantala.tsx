import React from 'react';
import Header from './header/Header';
import './Plantala.css';

function Plantala() {
  return (
    <div className="plantala">
      <Header />
    </div>
  );
}

export default Plantala;
