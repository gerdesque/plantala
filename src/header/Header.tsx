import React from 'react';
import logo from '../logo.png';
import './Header.css';

const Header = () => 
  <div className="app-header">  
    <img src={logo} className="app-logo" alt="logo" />
    <h1>Plantala</h1>
  </div>

export default Header;
