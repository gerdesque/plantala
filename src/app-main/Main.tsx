import React from 'react';
import { Button, Container, Grid, Typography } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import { IPlant } from '../plant/Plant';
import Landing from '../landing/Landing';
import PlantList from '../plant-list/PlantList';
import { Action } from '../app-plantala/Plantala';

export enum ButtonValue {
  Start = "Start",
  Select = "Zur Auswahl hinzufügen",
  Done = "Fertig",
  Again = "Neues Plantala erstellen"
}

export enum HeaderValue {
  Start = "Erstelle dein Mandala!",
  Select = "Wähle bis zu 5 Karten aus",
  Done = "Gestalte dein Plantala",
  Again = "Teile dein Plantala"
}

interface IMainProps {
  plants: IPlant[],
  action: Action,
  setAction: any
}

const useStyles = makeStyles((theme) => ({
  heroContent: {
    backgroundColor: theme.palette.background.default,
    padding: theme.spacing(2),
  },
  content: {
    paddingTop: theme.spacing(4),
  },
  main: {
    backgroundColor: theme.palette.primary.main,
  },
}));

export default function Main({ plants, action, setAction }: IMainProps) {
  const classes = useStyles();

  return (
    <main className={classes.main}>
      <Container maxWidth="md" className={classes.heroContent}>
        <Typography variant="h4" align="center" color="textPrimary" gutterBottom>
          {HeaderValue[action]}
        </Typography>
        <div>
          <Grid container spacing={2} justify="center">
            <Grid item md={3}>
              <Button variant="contained" color="secondary" fullWidth onClick={setAction}>
                {ButtonValue[action]}
              </Button>
            </Grid>
          </Grid>
        </div>
      </Container>
      <Container className={classes.content} maxWidth="md">
        {action === Action.Start && <Landing /> }
        {action === Action.Select && <PlantList plants={plants} /> }
          {/* <Mandala /> */}
      </Container>
    </main>
  );
}
