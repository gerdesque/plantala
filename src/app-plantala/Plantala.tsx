import React, { Component } from 'react';
import Header from '../app-header/Header';
import Main from '../app-main/Main';
import Footer from '../app-footer/Footer';
import './Plantala.css';
import { plants } from '../plants';
import { IPlant } from '../plant/Plant';

export enum Action {
  Start = "Start",
  Select = "Select",
  Done = "Done",
  Again = "Again"
}

interface IPlantalaState {
  plants: IPlant[],
  selectedPlants: IPlant[],
  activePlant: IPlant,
  action: Action
}

class Plantala extends Component {
  state: IPlantalaState = {
    // values to be displayed in <Card />
    plants: plants,
    // plants selected for <AvatarGroup />
    selectedPlants: plants,
    // active plant for <Plant />
    activePlant: {} as IPlant,
    // action mode to be displayed in <Main />
    action: Action.Start,
  }

  setAction = () => {
    this.setState({action: Action.Select});
  }

  render = () => {
    const { plants, selectedPlants, action } = this.state;

    return (
      <div className="plantala">
        <Header />
        <Main
          plants={plants}
          action={action}
          setAction={this.setAction} />
        <Footer
          selectedPlants={selectedPlants} />
      </div>
    );
  }
}

export default Plantala;
