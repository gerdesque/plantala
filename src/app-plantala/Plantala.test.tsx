import React from 'react';
import { Button } from '@material-ui/core';
import { createShallow, createMount } from '@material-ui/core/test-utils';
import Plantala from './Plantala';
import Header from '../app-header/Header';
import Main from '../app-main/Main';
import Footer from '../app-footer/Footer';

describe('Plantala', () => {

  let shallow: any;
  let wrapper: any;

  beforeEach(() => {
    shallow = createShallow();
    wrapper = shallow(<Plantala />);
  });

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());

  it('should render a <div /> container', () => {
    expect(wrapper.find('div').length).toEqual(1);
  });

  it('should render the Header, Main and Footer Component', () => {
    expect(
      wrapper.containsAllMatchingElements([
        <Header />,
        <Main
          plants={wrapper.instance().state.plants} 
          action={wrapper.instance().state.action}
          setAction={wrapper.instance().setAction}
          setSelectedPlant={wrapper.instance().setSelectedPlant}
        />,
        <Footer 
          selectedPlants={wrapper.instance().state.selectedPlants}
        />,
      ])
    ).toEqual(true);
  });
});

describe('mounted Plantala', () => {
  let mount: any;
  let wrapper: any;

  beforeEach(() => {
    mount = createMount();
    wrapper = mount(<Plantala />);
  });

  it('calls setAction when an action <Button /> is clicked', () => {
    const spy = jest.spyOn(wrapper.instance(), 'setAction');
    wrapper.instance().forceUpdate();
    expect(spy).toHaveBeenCalledTimes(0);
    wrapper
      .find(Button)
      .first()
      .simulate('click');
    expect(spy).toHaveBeenCalledTimes(1);
  });
});
