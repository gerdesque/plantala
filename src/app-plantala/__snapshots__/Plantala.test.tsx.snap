// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plantala should render correctly 1`] = `
<div
  className="plantala"
>
  <Header />
  <Main
    action="Start"
    plants={
      Array [
        Object {
          "description": "Lehrtafel von Carl von Tubeuf",
          "name": "Lathraea squamaria",
          "source": "1.png",
        },
        Object {
          "name": "Oedogonium diplandrum Juranyi",
          "source": "2.png",
        },
        Object {
          "name": "Nicotiana Tabacum L.",
          "source": "3.png",
        },
      ]
    }
    setAction={[Function]}
  />
  <Footer
    selectedPlants={
      Array [
        Object {
          "description": "Lehrtafel von Carl von Tubeuf",
          "name": "Lathraea squamaria",
          "source": "1.png",
        },
        Object {
          "name": "Oedogonium diplandrum Juranyi",
          "source": "2.png",
        },
        Object {
          "name": "Nicotiana Tabacum L.",
          "source": "3.png",
        },
      ]
    }
  />
</div>
`;
